utils/utils/src/main/scala/pl/touk/nussknacker/engine/util/json/BestEffortJsonEncoder.scala

```diff
@@ -70,13 +71,15 @@ case class BestEffortJsonEncoder(
           case a: OffsetDateTime => Encoder[OffsetDateTime].apply(a)
           case a: UUID           => safeString(a.toString)
           case a: DisplayJson    => a.asJson
-          case a: scala.collection.Map[String @unchecked, _] => encodeMap(a.toMap)
-          case a: java.util.Map[String @unchecked, _]        => encodeMap(a.asScala.toMap)
-          case a: Iterable[_]                                => fromValues(a.map(encode))
-          case a: Enum[_]                                    => safeString(a.toString)
-          case a: java.util.Collection[_]                    => fromValues(a.asScala.map(encode))
-          case a: Array[_]                                   => fromValues(a.map(encode))
-          case _ if !failOnUnknown                           => safeString(any.toString)
+          case a: java.util.Map[_, _] if allTheKeysAreJavaMaps(a.keySet())       => encodeComplexJavaMap(a)
```

Comment:
What about the performance impact on the situation when you have a large maps with simple keys? It will be a much more probable situation than map with complex key